### PR TITLE
Text spotting 0004

### PR DESCRIPTION
**Pull request by Ilya-Krylov**

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
the same architecture that's why FLOPS and PARAMS numbers have not been changed.

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+The text-spotting-0004-detector model is a Mask-RCNN-based text detector with ResNet50 backbone and additional text features output.
+
+| Metric                                        | Value     |
+|-----------------------------------------------|-----------|
+| GFlops                                        | 184.495   |
+| MParams                                       | 27.010    |
+
+### Inputs
+
+1.	Name: `im_data` , shape: [1x3x768x1280]. An input image in the [1xCxHxW] format.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:37]: We have agreement to use such format for input/output shape and layout specification (and plan to align it for all models), so please keep it:
```suggestion
1.	Name: `im_data` , shape: `1, 3, 768, 1280`. An input image in the `1, C, H, W` format.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    The expected channel order is BGR.
+2.	Name: `im_info`, shape: [1x3]. Image information: processed image height,
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:39]:
```suggestion
2.	Name: `im_info`, shape: `1, 3`. Image information: processed image height,
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    processed image width, and processed image scale with respect to the original image resolution.
+
+### Outputs
+
+1.	Name: `labels`, shape: [100]. Contiguous integer class ID for every
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:44]:
```suggestion
1.	Name: `labels`, shape: `100`. Contiguous integer class ID for every
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    detected object, `0` is for text class.
+2.	Name: `boxes`, shape: [100x5]. Bounding boxes around every detected object
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:46]:
```suggestion
2.	Name: `boxes`, shape: `100, 5`. Bounding boxes around every detected object
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    in the (top_left_x, top_left_y, bottom_right_x, bottom_right_y, confidence) format.
+3.	Name: `masks`, shape: [100x28x28]. Text segmentation masks for every output bounding box.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:48]:
```suggestion
3.	Name: `masks`, shape: `100, 28, 28`. Text segmentation masks for every output bounding box.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+4.  Name: `text_features.0`, shape [100x64x28x28]. Text features that are fed to a text recognition head.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:49]:
```suggestion
4.  Name: `text_features.0`, shape `100, 64, 28, 28`. Text features that are fed to a text recognition head.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+The text-spotting-0004-recognizer-encoder model is a fully-convolutional encoder of text recognition head.
+
+| Metric                                        | Value     |
+|-----------------------------------------------|-----------|
+| GFlops                                        | 2.082     |
+| MParams                                       | 1.328     |
+
+### Inputs
+
+Name: `input` , shape: [1x64x28x28]. Text recognition features obtained from detection part.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:62]:
```suggestion
Name: `input` , shape: `1, 64, 28, 28`. Text recognition features obtained from detection part.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+### Outputs
+
+Name: `output`, shape: [1x256x28x28]. Encoded text recognition features.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:66]:
```suggestion
Name: `output`, shape: `1, 256, 28, 28`. Encoded text recognition features.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+## Decoder model specification
+
+| Metric                                        | Value     |
+|-----------------------------------------------|-----------|
+| GFlops                                        | 0.106     |
+| MParams                                       | 0.283     |
+
+### Inputs
+
+1.	Name: `encoder_outputs` , shape: [1x(28*28)x256]. Encoded text recognition features.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:77]:
```suggestion
1.	Name: `encoder_outputs` , shape: `1, (28*28), 256`. Encoded text recognition features.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+1.	Name: `prev_symbol` , shape: [1x1]. Index in alphabet of previously generated symbol.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:78]:
```suggestion
1.	Name: `prev_symbol` , shape: `1, 1`. Index in alphabet of previously generated symbol.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+1.	Name: `prev_hidden`, shape: [1x1x256]. Previous hidden state of GRU.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:79]:
```suggestion
1.	Name: `prev_hidden`, shape: `1, 1, 256`. Previous hidden state of GRU.
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+### Outputs
+
+1.	Name: `output`, shape: [1x38]. Encoded text recognition features. Indices starting from 2 correspond to symbols from the
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:83]:
```suggestion
1.	Name: `output`, shape: `1, 38`. Encoded text recognition features. Indices starting from 2 correspond to symbols from the
```

File: models/intel/text-spotting-0004/description/text-spotting-0004.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+alphabet. The 0 and 1 are special Start of Sequence and End of Sequence symbols correspondingly.
+1.	Name: `hidden`, shape: [1x1x256]. Current hidden state of GRU.
```
Comment (eaidova) [inline models/intel/text-spotting-0004/description/text-spotting-0004.md:85]:
```suggestion
1.	Name: `hidden`, shape: `1, 1, 256`. Current hidden state of GRU.
```

**Review by eaidova — APPROVED** [review]

**Review by Wovchena — APPROVED** [review]

**Comment by eaidova** [comment]
jenkins please retry a build

**Review by vladimir-dudnik — APPROVED** [review]

**Review by IRDonch — APPROVED** [review]

**Pull request merged**